app/lib/actions/pulp/repository/remove_units.rb

```diff
@@ -0,0 +1,33 @@
+module Actions
+  module Pulp
+    module Repository
+      class RemoveUnits < Pulp::AbstractAsyncTask
+        input_format do
+          param :repo_id
+          param :contents
+        end
+
+        def invoke_external_task
+          repo = ::Katello::Repository.find(input[:repo_id])
+          repo_content_types = ::Katello::RepositoryTypeManager.find(repo.content_type).content_types
+          if(input[:contents])
+            user_removable_content_type = repo_content_types.first
+            units = user_removable_content_type.model_class.where(:id => input[:contents])
```

Comment:
`repo_content_types.first` should return content_type based on priority. Checked the registration and all user removable content types are either single types or priority 1.